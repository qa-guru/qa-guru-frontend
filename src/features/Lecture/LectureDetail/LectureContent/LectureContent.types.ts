export interface ILectureContent {
  content: any;
}
