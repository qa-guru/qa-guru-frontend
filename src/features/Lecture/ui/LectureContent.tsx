import React from "react";
import { Box, CardMedia, Link, Paper, Stack, Typography } from "@mui/material";

interface ILectureContent {
  contentLecture: any;
}

const style = {
  paper: { padding: "20px" },
  box: {
    overflow: "hidden",
    paddingBottom: "40.25%",
    position: "relative",
    height: 0,
  },
  iframe: {
    left: 0,
    top: 0,
    height: "100%",
    width: "70%",
    position: "absolute",
  },
};

const LectureContent: React.FC<ILectureContent> = (props) => {
  const { contentLecture } = props;

  return (
    <>
      <Typography mb="15px" pt="40px" variant="h4">
        Материалы урока
      </Typography>
      <Paper sx={style.paper}>
        <Stack spacing={0.7}>
          {contentLecture.map((item: any, index: any) => {
            const { value, url, type } = item;

            switch (type) {
              case "title":
                return (
                  <Typography key={index} variant="h6">
                    {value}
                  </Typography>
                );
              case "text":
                return (
                  <Typography key={index} variant="subtitle1">
                    {value}
                  </Typography>
                );
              case "video":
                return (
                  <Box key={index}>
                    <Typography variant="h6">{value}</Typography>
                    <CardMedia image={url} />
                    <Box sx={style.box}>
                      <iframe
                        // @ts-ignore
                        style={style.iframe}
                        src={`https://www.youtube.com/embed/${url}`}
                        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
                        allowFullScreen
                        title="Embedded youtube"
                      />
                    </Box>
                  </Box>
                );
              case "link":
                return (
                  <Link key={index} underline="hover" href={url}>
                    {value}
                  </Link>
                );
            }
          })}
        </Stack>
      </Paper>
    </>
  );
};

export default LectureContent;
