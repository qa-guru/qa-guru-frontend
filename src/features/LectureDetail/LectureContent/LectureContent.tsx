import React from "react";
import { Paper, Stack, Typography } from "@mui/material";
import { ILectureContent } from "./LectureContent.types";
import Content from "../../../shared/Content";

const style = {
  paper: { padding: { xs: "15px", md: "20px" } },
};

const LectureContent: React.FC<ILectureContent> = ({ content }) => {
  return (
    <>
      <Typography mb="15px" pt="30px" variant="h4">
        Материалы урока
      </Typography>
      <Paper sx={style.paper}>
        <Stack spacing={0.5}>
          <Content content={content} />
        </Stack>
      </Paper>
    </>
  );
};

export default LectureContent;
