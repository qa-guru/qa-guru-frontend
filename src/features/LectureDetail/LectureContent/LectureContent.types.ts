export interface ILectureContent {
  content: Array<{
    __typename?: "LectureContentDto";
    type?: string | null;
    value?: string | null;
    url?: string | null;
  } | null> | null;
}
