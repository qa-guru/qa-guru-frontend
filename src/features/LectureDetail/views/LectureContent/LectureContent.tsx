import React from "react";
import { Paper, Stack, Typography } from "@mui/material";
import { ILectureContent } from "./LectureContent.types";
import ContentSerialization from "../../../../shared/Serializers/ContentSerialization";

const style = {
  paper: { padding: { xs: "15px", md: "20px" } },
};

const LectureContent: React.FC<ILectureContent> = ({ content }) => {
  return (
    <>
      <Typography mb="15px" pt="30px" variant="h4">
        Материалы урока
      </Typography>
      <Paper sx={style.paper}>
        <Stack spacing={0.5}>
          <ContentSerialization content={content} />
        </Stack>
      </Paper>
    </>
  );
};

export default LectureContent;
