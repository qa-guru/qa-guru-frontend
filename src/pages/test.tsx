import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
