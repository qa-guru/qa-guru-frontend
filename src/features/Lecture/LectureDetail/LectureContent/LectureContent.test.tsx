import { MemoryRouter } from "react-router-dom";
import LectureContent from "./LectureContent";
import { render } from "../../../../test/utilsTest";

const mockContentLecture: any = [];

describe("LectureContent", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LectureContent content={mockContentLecture!} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
