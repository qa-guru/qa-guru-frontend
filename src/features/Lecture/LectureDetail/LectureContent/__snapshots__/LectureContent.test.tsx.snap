// Vitest Snapshot v1

exports[`LectureContent > the component is render 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root MuiTypography-h4 css-1t14ubg-MuiTypography-root"
  >
    Материалы урока
  </h4>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1u0eh3h-MuiPaper-root"
  >
    <div
      class="css-j722dc-MuiStack-root"
    />
  </div>
</DocumentFragment>
`;
